ocaml/xenopsd/xc/service.ml

```diff
@@ -376,31 +376,32 @@ module Qemu = struct
 
   let stop ~xs ~qemu_domid domid =
     let file_path = pidfile_path domid in
-    match pid ~xs domid with
-    | None ->
-        Unixext.unlink_safe file_path
-    | Some pid ->
-        let xenstore_path = pidxenstore_path domid in
-        let best_effort = Xenops_utils.best_effort in
-        let really_kill = Xenops_utils.really_kill in
-        debug "qemu-dm: stopping qemu-dm with SIGTERM (domid = %d)" domid ;
-        best_effort
-          "signalling that qemu is ending as expected, mask further signals"
-          (fun () -> SignalMask.set signal_mask domid
-        ) ;
-        best_effort "killing qemu-dm" (fun () -> really_kill pid) ;
-        best_effort "removing qemu-pid from xenstore" (fun () ->
-            xs.Xs.rm xenstore_path
-        ) ;
-        best_effort "unmasking signals, qemu-pid is already gone from xenstore"
-          (fun () -> SignalMask.unset signal_mask domid
-        ) ;
-        best_effort "removing device model path from xenstore" (fun () ->
-            xs.Xs.rm (Device_common.device_model_path ~qemu_domid domid)
-        ) ;
-        best_effort (Printf.sprintf "removing %s" file_path) (fun () ->
-            Unix.unlink file_path
-        )
+    let pid = pid ~xs domid |> Option.value ~default:(-1) in
```

Comment:
I can't see why this `-1` is being introduced.
Wouldn't the intrinsic `Option` work well here?

Reply:
It's used in below code:
`if pid <> -1 `

Reply:
Thanks, fixed.